client/my-sites/domains/domain-management/settings/cards/contact-verification-card.tsx

```diff
@@ -0,0 +1,212 @@
+					{ contactInformation.fname } { contactInformation.lname }
+				</p>
+				{ contactInformation.org && <p>{ contactInformation.org }</p> }
+				<p>{ contactInformation.email }</p>
+				<p>{ contactInformation.sa1 }</p>
+				{ contactInformation.sa2 && <p>{ contactInformation.sa2 }</p> }
+				<p>
+					{ contactInformation.city }
+					{ contactInformation.sp && <span>, { contactInformation.sp }</span> }
+					<span> { contactInformation.pc }</span>
+				</p>
+				<p>{ contactInformation.country_code }</p>
+				<p>{ contactInformation.phone }</p>
+				{ contactInformation.fax && <p>{ contactInformation.fax }</p> }
+			</div>
+		);
+	};
+
+	const renderDescription = () => {
+		const { contactInformationUpdateLink } = props;
+
+		return (
+			<div>
+				<p>
+					{ translate(
+						'Nominet, the organization that manages .uk domains, requires us to verify the contact information of your domain.'
+					) }
+				</p>
+				<p>{ translate( 'This is your current contact information:' ) }</p>
+				{ renderContactInformation() }
+				<p>
+					{ translate(
+						'Please verify that the above infomration is correct and either {{a}}update it{{/a}} or provide a photo of a document on which the above name and address are clearly visible. Some of the accepted documents are:',
+						{
+							components: {
+								a: contactInformationUpdateLink ? (
+									<a href={ contactInformationUpdateLink } />
+								) : (
+									<span />
+								),
+							},
+						}
+					) }
+				</p>
+				<ul>
+					<li>{ translate( "Valid drivers' license" ) }</li>
+					<li>{ translate( 'Valid national ID cards (for non-UK residents)' ) }</li>
+					<li>{ translate( 'Utility bills (last 3 months)' ) }</li>
+					<li>{ translate( 'Bank statement (last 3 months)' ) }</li>
+					<li>{ translate( 'HMRC tax notification (last 3 months)' ) }</li>
+				</ul>
+				<p>
+					{ translate(
+						'Click on the button below to upload one or more documents and then click on the "Submit" button. You can upload images and/or PDF files up to 5MB each.'
+					) }
+				</p>
+			</div>
+		);
+	};
+
+	const handleFilesSelected = ( files: FileList ) => {
+		setSelectedFiles( files );
+		setError( false );
+	};
+
+	const renderSelectFilesButton = () => {
+		return (
+			<div>
+				<FilePicker multiple accept="image/*,.pdf" onPick={ handleFilesSelected }>
+					<Button disabled={ submitting || submitted }>{ translate( 'Select files' ) }</Button>
+				</FilePicker>
+			</div>
+		);
+	};
+
+	const renderSelectedFileList = () => {
+		if ( ! selectedFiles || selectedFiles.length === 0 ) {
+			return <p>{ translate( 'No selected files yet' ) }</p>;
+		}
+
+		const fileList = [ ...selectedFiles ].map( ( file: File ) => (
+			<li key={ file.name }>{ file.name }</li>
+		) );
+		return (
+			<div className="ownership-verification-card__selected-files">
+				<p>{ translate( 'Selected files:' ) }</p>
+				<ul>{ fileList }</ul>
+			</div>
+		);
+	};
+
+	const submitFiles = () => {
+		wpcom.req.post(
```

Comment:
Also, [according to the MDN documentation](https://developer.mozilla.org/en-US/docs/Web/API/FormData/append#:~:text=As%20with%20regular%20form%20data%2C%20you%20can%20append%20multiple%20values%20with%20the%20same%20name%3A) - which I just tested above, it's fine to use the same name for different files, so `files[]` should work. 

I know it's also highly opinionated, but I had to postfix the key with `[]`, so that PHP interprets it as an array.
If I don't add it, then it only a single file would be recognized, and the others would be ignored:

<img width=300 src="https://user-images.githubusercontent.com/18705930/229022217-0ebc377f-da95-4bb2-b00e-454f4285204e.png"/>

Reply:
@rafaelgalani thanks a lot for the detailed explanation! I wasn't sure if the way I was doing file uploads was the right way 😬